cli/src/main/java/org/jboss/as/cli/impl/ReadlineConsole.java

```diff
@@ -694,67 +718,73 @@ public void start() throws IOException {
+            }
+            // Console could have been closed during a command execution.
+            if (!closed) {
+                getReadLine().readline(connection, prompt, line -> {
+                    // All commands can lead to prompting the user. So require
+                    // to be executed in a dedicated thread.
+                    // This can happen if a security configuration occurs
+                    // on the remote server.
+                    LOG.tracef("Executing command %s in a new thread.", line);
+                    if (line == null || line.trim().length() == 0 || handleAlias(line)) {
+                        loop();
+                        return;
+                    }
+                    executor.submit(() -> {
+                        Consumer<Signal> handler = connection.getSignalHandler();
+                        Thread callingThread = Thread.currentThread();
+                        connection.setSignalHandler((signal) -> {
+                            // Interrupting the current command thread.
+                            switch (signal) {
+                                case INT: {
+                                    LOG.tracef("Interrupting command: %s", line);
+                                    callingThread.interrupt();
+                                }
                             }
+                        });
+                        try {
+                            callback.accept(line);
+                        } catch (Throwable thr) {
+                            connection.write("Unexpected exception");
+                            thr.printStackTrace();
+                        } finally {
+                            // The current thread could have been interrupted.
+                            // Clear the flag to safely interact with aesh-readline
+                            Thread.interrupted();
+                            connection.setSignalHandler(handler);
+                            LOG.tracef("Done Executing command %s", line);
+                            loop();
                         }
                     });
-                    try {
-                        callback.accept(line);
-                    } finally {
-                        // The current thread could have been interrupted.
-                        // Clear the flag to safely interact with aesh-readline
-                        Thread.interrupted();
-                        connection.setSignalHandler(handler);
-                        LOG.tracef("Done Executing command %s", line);
-                        loop();
-                    }
-                });
-            }, completions, preProcessors, readlineHistory, null, READLINE_FLAGS);
+                }, completions, preProcessors, readlineHistory, null, READLINE_FLAGS);
+            }
+        } catch (Exception ex) {
+            connection.write("Unexpected exception");
+            ex.printStackTrace();
```

Comment:
Are we trying to eliminate these?

Reply:
@bstansberry , you mean the printStackTrace? This one should be printed to the user, it means a bug in CLI for which the stack should be reported.

Reply:
Ok, thanks. That's fine.